design/dsl/endpoint_test.go

```diff
@@ -148,9 +146,17 @@ func assertEndpointMetaData(t *testing.T, actual design.MetadataExpr, expected d
 }
 
 func assertEndpointRequestResponse(t *testing.T, assertType string, actual design.DataType, expected *design.UserTypeExpr) {
+
 	ut, ok := actual.(*design.UserTypeExpr)
-	if !ok || ut == nil {
-		t.Errorf("expected endpoint %s to be a *UserTypeExpr but got %v", assertType, reflect.TypeOf(ut))
+	if !ok {
+		//test on the DataType
+		if actual.Kind() != expected.Type.Kind() {
```

Comment:
I don't think this test is needed?  The above already checked that `actual` is not a `design.UserTypeExpr`. The error message could be simplified too as per the previous comment.